[Irminrics] iP

angjunkang opened the pull request:
# Dude[](https://github.com/Irminrics/ip)

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Dude frees your mind of having to remember things you need to do. It's,
-   text-based
-   easy to learn
-   FAST  _SUPER_  FAST to use


All you need to do is,
1.  download it from  [here](https://github.com/Irminrics/ip/releases/tag/A-Jar).
2.  double-click it.
3.  add your tasks.
4.  let it manage your tasks for you 😉

And it is  **FREE**!

Features:

- [X]   Managing tasks
- [X]   Managing deadlines
- [X]   Finding Keywords
- [ ]   Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the  `main`  method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

angjunkang committed:
The birth of Singaporean Dude

angjunkang committed:
Level-1 - Formatting the dude

angjunkang committed:
Level-1 - Formatting the dude 2

angjunkang committed:
List has been added to Dude's inventory

angjunkang committed:
Mark Level-3

angjunkang committed:
Dude has introduced 3 events to the list

angjunkang committed:
Dude cannot mark with 0 task

angjunkang committed:
Dude obtained the power of automated testing

angjunkang committed:
Dude loves longer dotted lines and exception handling!

angjunkang committed:
Dude has mastered the art of deletion

angjunkang committed:
Dude has obtained enumerations

angjunkang committed:
Change Dude's structure

angjunkang committed:
Dude has mastered the art of using file storage

angjunkang committed:
Cleaning up dude's output format

angjunkang committed:
Added validation to Dude's deletion task

angjunkang committed:
Dude is using LocalDateTime for task details

angjunkang committed:
Merge branch 'master' into branch-Level-7
Details: Dude is using LocalDateTime for task details

angjunkang committed:
Merge branch 'master' into branch-Level-8
Details: Added validation to Dude's deletion task

angjunkang committed:
Merge branch 'branch-Level-8'
Conflicts to be resolved in the next commit: Event.java, Dude.java

angjunkang committed:
Dude has resolved merge conflict  for branch-Level-8 into master

angjunkang committed:
Dude fixed file write and read after merging with LocalDateTime type.

angjunkang committed:
Dude underwent major update for the sake of OOP.
He is now lacking validations for commands and will be fixed soon.

angjunkang committed:
Removing Dude's unnecessary files and packaging

angjunkang committed:
Merge branch 'add-gradle-support'

angjunkang committed:
Dude fixed his annoying warnings

angjunkang committed:
Dude combined other dude exceptions into one single custom exception

angjunkang committed:
Bug fix for typo in Dude's custom exception

angjunkang committed:
Add validation to commands

angjunkang committed:
Add JUnit test for tasks

angjunkang committed:
Add invalid command validation

angjunkang committed:
Add MANIFEST.MF to package application as JAR file

angjunkang committed:
Add data/task.txt into .gitignore

angjunkang committed:
Add JavaDoc comments to the code

angjunkang committed:
Tweak the code to comply with a coding standard

angjunkang committed:
Add find command

angjunkang committed:
Merge branch 'branch-A-CodingStandard'

angjunkang committed:
Merge branch 'branch-A-JavaDoc'

angjunkang committed:
Fix merge conflict code to comply with a coding standard

angjunkang committed:
Merge branch 'branch-Level-9'

angjunkang committed:
Fix merge conflict with Level-9

angjunkang committed:
Fix merge conflict with Level 9

Fix the change in method name from removeTaskCount to decreaseTaskCount

angjunkang committed:
Fix commands validation for deadline and event

angjunkang committed:
Fix scanner issue and pathing for runtest.bat

angjunkang committed:
Update README.md

angjunkang committed:
Add and use checkStyle to detect coding style violations

angjunkang committed:
Use checkStyle to detect coding style violations

angjunkang committed:
Add program loading error message

angjunkang committed:
Fix unmark error and checkstyle issues

angjunkang committed:
Add a GUI to Dude using the JavaFX technology

angjunkang committed:
Merge branch 'branch-Level-10'

angjunkang committed:
Add custom user icon, chat borders and command to exit the program

angjunkang committed:
Change find command to check for multiple keywords

angjunkang committed:
Merge branch 'branch-A-Varargs'

LeirdaKJW commented on src/main/java/dude/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+    /**
+     * Enums for different possible command types
+     */
+    public enum CommandType {
+        LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND, BYE
+    }
```
I like how you used enums to ensure type safety of the commands.

LeirdaKJW commented on src/main/java/dude/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+     * @param command Command input provided by user.
+     * @throws DudeException  If command is not valid.
+     */
+    public static void validate(CommandType commandType, String command) throws DudeException {
```
I think this is a smart way to handle the input exceptions!

LeirdaKJW commented on src/main/java/dude/Dude.java:
```diff
@@ -0,0 +1,71 @@
+        }
+    }
+
+    /**
+     * Runs Dude program.
+     */
```
I feel like this could be a bit more descriptive.

LeirdaKJW commented on src/main/java/dude/Dude.java:
```diff
@@ -0,0 +1,71 @@
+/**
+ * Handles the entire dude program
+ */
+public class Dude {
```
Clean and easy to understand!

LeirdaKJW reviewed (commented):
Overall, super clean code and easy-to-understand code! Basically, no coding standard violations from what I see.😄

angjunkang committed:
Fix issue with JavaFX label ellipse

angjunkang committed:
Fix width of dialog box

angjunkang committed:
Add assertions

Dude has no assertions

Assertions are used to define assumptions about the program state so that the runtime can verify them.

Let's add assertions to check for functions preconditions/parameter

This ensures that the inputs for functions are validated and ensured for correctness

angjunkang committed:
Merge pull request #1 from Irminrics/branch-A-Assertions

Add assertions

angjunkang committed:
Improve Code Quality

Dude has no usage of guard clause

Guard clause helps to keep the main/happy path unindented

Let's add guard clause to the execute function in MarkCommand and UnmarkCommand

This deals with unusual conditions as soon as they are detected so that
the reader does not have to remember them for long

angjunkang committed:
Merge pull request #2 from Irminrics/!branch-A-CodeQuality

Improve Code Quality

angjunkang committed:
Set up CI

angjunkang committed:
Add undo feature

Users of Dude are unable to undo their previous command.

This is a quality of life feature which allows them to undo mistakes.

Let's add a feature which saves the previous state of task list for the
users to undo upon a new command.

angjunkang committed:
Merge pull request #3 from Irminrics/branch-C-Undo

Add undo feature

angjunkang committed:
Change workflow folder to workflows

angjunkang committed:
Fix exit command

angjunkang committed:
Add a representative screenshot of application

angjunkang committed:
Fix data validation bug for deadline and event

angjunkang committed:
Add a user guide

angjunkang committed:
Fix checkstyle violations

angjunkang committed:
Change main class name for application in build.gradle

angjunkang committed:
Fix javadocs coding standard raised by nus-se bot

angjunkang committed:
Change representative screenshot name

angjunkang committed:
Fix typo in user guide

angjunkang committed:
Fix delete task response bug

angjunkang committed:
Add validation for index less than or equals to 0 for Delete, Mark and Unmark commands
